### PR TITLE
fix(gatsby-source-strapi): `options_` bug in source-strapi@3.0.0

### PR DESCRIPTION
**Pull request by laurenskling**
Hey @moonmeister ,
found 2 little bugs. 

First off a `options_` was left over after rewriting some prettier automated fixes.
<details>
<summary>Moving rest to #349</summary>

Second, the plugin doesn't want to run without a `index.js` for me 🤷
Getting:

```
Couldn't find the "gatsby-source-strapi" plugin declared in "/Users/laurenskling/sites/gatsby-monorepo/packages/gatsby-theme-core/gatsby-config.js".

Tried looking for an installed package in the following paths:
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby/dist/bootstrap/load-themes/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby/dist/bootstrap/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby/dist/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/node_modules/gatsby-source-strapi
 - /Users/laurenskling/node_modules/gatsby-source-strapi
 - /Users/node_modules/gatsby-source-strapi
 - /node_modules/gatsby-source-strapi
```

Adding a `index.js` noop fixes it (source-wordpress also has it). I cant empty a empty file due to linting rules tho :( thats why i added the log statement. If you have another trick its much welcomed. Cant seem to ignore the lint error--
</details>

### CODE REVIEW

**Comment by moonmeister** [comment]
What's the error with index.js? Cause none of the other plugins have that file.

File: packages/gatsby-source-strapi/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+console.log("noop");
```
Comment (moonmeister) [inline packages/gatsby-source-strapi/index.js:1]: This can be a comment, not actual code.

**Review by moonmeister — CHANGES_REQUESTED** [review]
Why is yarn.lock changing? Nothing here suggests it should.

File: packages/gatsby-source-strapi/index.js (hunk 1 above)
Comment (laurenskling) [reply packages/gatsby-source-strapi/index.js:1]: Well, then lint will crash on [no-empty-file](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-empty-file.md)

**Comment by laurenskling** [comment]
> What's the error with index.js? Cause none of the other plugins have that file.

As I said, it gives 

```
Couldn't find the "gatsby-source-strapi" plugin declared in "/Users/laurenskling/sites/gatsby-monorepo/packages/gatsby-theme-core/gatsby-config.js".

Tried looking for an installed package in the following paths:
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby/dist/bootstrap/load-themes/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby/dist/bootstrap/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby/dist/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/gatsby-monorepo/node_modules/gatsby-source-strapi
 - /Users/laurenskling/sites/node_modules/gatsby-source-strapi
 - /Users/laurenskling/node_modules/gatsby-source-strapi
 - /Users/node_modules/gatsby-source-strapi
 - /node_modules/gatsby-source-strapi
```

I don't know why. `filesystem` and `wordpress` also have it. I saw `s3` doesn't have it. No idea how Gatsby tries to find the plugin. I don't see the difference.

File: packages/gatsby-source-strapi/index.js (hunk 1 above)
Comment (laurenskling) [reply packages/gatsby-source-strapi/index.js:1]: Damn, the yarn lock is there for my yarn lock fix of last time i did on the main branch, forgot to pull the main branch before making this branch. Will fix it tomorrow, need to run now.

**Comment by laurenskling** [comment]
@moonmeister I forced pushed a new clean commit

**Comment by moonmeister** [comment]
So reality is we shouldn't need this, Gatsby doesn't care and we have other plugins without these. What's different about this? I'm happy to help, may have time tomorrow, but I'd start by trying to reproduce on a clean site and go from there. 

Basically, I want to get to the root cause, non just resolve the symptom.

**Comment by moonmeister** [comment]
So reality is we shouldn't need this, Gatsby doesn't care and we have other plugins without these. What's different about this? I'm happy to help, may have time tomorrow, but I'd start by trying to reproduce on a clean site and go from there. 

Basically, I want to get to the root cause, non just resolve the symptom. 


Update: I also don't want you to feel like I'm trying to be controlling, If you think this is the right move then I'll step aside, but it's not adding up to me, and I'd recommend we figure out why.

File: packages/gatsby-source-strapi/index.js (hunk 1 above)
Comment (moonmeister) [reply packages/gatsby-source-strapi/index.js:1]: > Well, then lint will crash on [no-empty-file](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-empty-file.md)

If we do add this we'll add an ignore.

**Comment by moonmeister** [comment]
I'm wondering if https://github.com/gatsby-uc/plugins/blob/main/packages/gatsby-source-strapi/package.json#L11-L13 might be doing it...that config is in no other package.

**Comment by moonmeister** [comment]
Okay, after some reading we have a packaging problem, what we think is in the package isn't. Check out the package in `node_modules` and see what's actually in it, I suspect we're missing files.

**Comment by laurenskling** [comment]
> I'm wondering if https://github.com/gatsby-uc/plugins/blob/main/packages/gatsby-source-strapi/package.json#L11-L13 might be doing it...that config is in no other package.

I was thinking the same, but removing that in my local project doesn't make a difference

**Comment by laurenskling** [comment]
> So reality is we shouldn't need this, Gatsby doesn't care and we have other plugins without these. What's different about this? I'm happy to help, may have time tomorrow, but I'd start by trying to reproduce on a clean site and go from there.
> 
> Basically, I want to get to the root cause, not just resolve the symptom.

I'd like to know as well, just cannot find anything on the matter. Just know that all gatsby official sources have them as well: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-faker/index.js

**Comment by laurenskling** [comment]
It's crashing here without the index.js:
https://github.com/gatsbyjs/gatsby/blob/26f2b72a63b29ef8425cd33ae70708279cd4af87/packages/gatsby/src/bootstrap/load-themes/index.ts#L63

**Comment by laurenskling** [comment]
It doesn't resolve here: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/bootstrap/load-themes/index.ts#L41

**Comment by laurenskling** [comment]
I don't really know a lot about node resolving to understand this. I guess easiest is to just add the noop index.js

**Comment by timeswind** [comment]
These bugs are critical, would affect user reading tutorials from Gatsby docs and updating Gatsby from v4 to v5.

**Review by moonmeister — APPROVED** [review]

**Pull request merged**